llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -1444,6 +1444,12 @@ class TargetInstrInfo : public MCInstrInfo {
     return false;
   }
 
+  /// Returns true if MI is a far call.
```

Comment:
Is this (or should it be) an x86 thing?

Reply:
far calls are an x86 thing yeah, but I think I'm forced to implement it here in the generic base class, since the asmprinter is unaware of platform details. We override this implementation in the sub-class. I'm happy for the base class to return `false` for non-x86 architectures.

Reply:
Or, at least, `#ifdef` these functions for x86?

Reply:
We could, but it hardly seems worth it, especially since (in theory, if you enjoy pain), you could be cross-compiling.

Reply:
Then I think we should at least do something like `YKTODO: x86 only` to remind our future selves that these aren't completely generic.